src/lib/shadow-shim-helper-rs/src/shim_event.rs

```diff
@@ -3,6 +3,17 @@ use vasi::VirtualAddressSpaceIndependent;
 
 use crate::syscall_types::{ForeignPtr, SyscallArgs, SyscallReg, UntypedForeignPtr};
 
+#[derive(Copy, Clone, Debug, VirtualAddressSpaceIndependent)]
+#[repr(C)]
+/// Data for [`ShimEventToShim::Syscall`] and [`ShimEventToShadow::Syscall`]
```

Comment:
I see this used in `ShimEventToShim::StartRes`, but not `ShimEventToShim::Syscall` or `ShimEventToShadow::Syscall`?